### PR TITLE
Improved client timeout handling

### PR DESCRIPTION
**Pull request by ricsiLT**
Make timeout configurable
Add some descriptions to enum cases

### CODE REVIEW

**Comment by Gsantomaggio** [comment]
Can you please update the documentation? https://github.com/rabbitmq/rabbitmq-stream-dotnet-client#invalidate-messages. 

Maybe explain the ClientSide exception vs server-side exception (`ClientTimeoutError`)

File: RabbitMQ.Stream.Client/Reliable/ReliableProducer.cs [hunk 1]
```diff
@@ -17,6 +17,7 @@ public record ReliableProducerConfig
     public Func<MessagesConfirmation, Task> ConfirmationHandler { get; init; }
     public string ClientProvidedName { get; set; } = "dotnet-stream-rproducer";
     public IReconnectStrategy ReconnectStrategy { get; set; } = new BackOffReconnectStrategy();
+    public TimeSpan TimeoutMessageAfter { get; init; } = TimeSpan.FromSeconds(3);
```
Comment (Gsantomaggio) [inline RabbitMQ.Stream.Client/Reliable/ReliableProducer.cs:20]: Should we add a sort of validation to avoid to insert wrong values like `0` or `Hours` ?

Comment (ricsiLT) [reply RabbitMQ.Stream.Client/Reliable/ReliableProducer.cs:20]: No idea - 0 doesn't make sense for sure, so we can definitely check for that. For other values - other than just wasting space in cache, what's the danger on waiting for days on end? :D

Comment (Gsantomaggio) [reply RabbitMQ.Stream.Client/Reliable/ReliableProducer.cs:20]: > what's the danger on waiting for days on end? :

The unconfirmed list is in memory. Just a memory problem. I'd say!

**Comment by Gsantomaggio** [comment]
Thank you @ricsiLT

**Pull request merged**